apm-lambda-extension/extension/apm_server.go

```diff
@@ -20,21 +20,37 @@ package extension
 import (
 	"bytes"
 	"compress/gzip"
+	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"log"
+	"math"
+	"math/rand"
 	"net/http"
 	"sync"
+	"time"
 )
 
 var bufferPool = sync.Pool{New: func() interface{} {
 	return &bytes.Buffer{}
 }}
 
+var apmServerTransportFailing bool
+var apmServerReconnectionCount int
+var apmServerBackoffTimer *time.Timer
+
+func InitApmServerTransportStatus() {
+	apmServerTransportFailing = false
+}
```

Comment:
Not sure what's idiomatic in Go, but there are `init()` functions that are executed automatically: https://go.dev/ref/spec#Package_initialization

Reply:
That is the most idiomatic solution, but I ended up needing an explicit init function in order to use unit tests (`init()` is only used once upon package initialization, which breaks unit tests when those are run sequentially)